src/julia/parser.jl

```diff
@@ -2197,7 +2197,15 @@ function parse_function_signature(ps::ParseState, is_function::Bool)
             is_empty_tuple = peek(ps, skip_newlines=true) == K")"
             opts = parse_brackets(ps, K")") do had_commas, had_splat, num_semis, num_subexprs
                 _parsed_call = was_eventually_call(ps)
-                _needs_parse_call = peek(ps, 2) ∈ KSet"( ."
+                # Check if we should skip newlines - only for specific cases
+                # where we have a single type annotation like (::T)
+                _skip_newlines = !had_commas && num_subexprs == 1 && 
```

Comment:
I'll be honest in that I don't really know what is going on here...

Reply:
I think the correct solution is to move the `_needs_parse_call` outside the do block - it seems weird that it would affect `needs_parameters`, because if there are parameters, the peek will never actually see the parenthesis.